moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -732,9 +732,11 @@ bool PlanningSceneMonitor::newPlanningSceneMessage(const moveit_msgs::msg::Plann
     if (!scene.is_diff && parent_scene_)
     {
       // if the scene does not contain a robot state then use the latest value from the current_state_monitor_
-      // so that we are not updating the scene with stale values for the robots state.
-      if (scene.robot_state.joint_state.name.empty())
+      // so that we are not updating the scene with stale values for the robot's state.
+      if (scene.robot_state.joint_state.name.empty() && current_state_monitor_->haveCompleteState())
```

Comment:
You also need an `if (current_state_monitor_)` -- sorry that my comments were too scattered

Reply:
Sorry I noticed that while I was writing my response (below).